### PR TITLE
[ENH] AptaMCTS Implementation

### PR DESCRIPTION
**Pull request by aditi-dsi**
#### Reference Issues/PRs
refs #21 
#### What does this implement/fix? Explain your changes.

This PR implements the cleaned version of AptaMCTS algorithm. Specifically, it adds:

- The iCTF sequence encoder utilities for RNA and proteins.

- A scikit-learn compatible Random Forest classifier (with proper __sklearn_tags__ for binary classification).

- An end-to-end sklearn pipeline connecting feature encoder and classification.

- The experiment wrapper that allows the MCTS engine to evaluate sequence candidates using the AptaMCTS scoring pipeline.

#### Did you add any tests for the change?

Yes. Adds a test suite in `pyaptamer/aptamcts/tests/test_aptamcts.py`.

#### Any other comments?
Checklist items:

- [x] Initial skeleton and docstrings for `_feature_classifier.py`
- [x] Initial skeleton and docstrings for `_pipeline.py`
- [x] Implement each module/classes.
- [x] Write `experiments/_aptamer_aptamcts.py`
- [x] Add corresponding tests

#### PR checklist

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.

### CODE REVIEW

**Comment by aditi-dsi** [comment]
@NennoMP this PR is ready for review.

**Comment by NennoMP** [comment]
Do not request a review if tests are failing.

**Comment by aditi-dsi** [comment]
@NennoMP My bad, I accidentally missed running the doctests locally. Fixed the failures now, please feel free to re-run.

File: pyaptamer/aptamcts/_feature_classifier.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+
+class AptaMCTSClassifier(ClassifierMixin, BaseEstimator):
+    """
+    Random Forest-based binary classifier used as a scoring function for AptaMCTS.
+
+    This model applies a sklearn Random Forest classifier to predict aptamer-protein
+    interactions. It only accepts numerical features. RNA and protein sequences
+    must already be converted into numbers (e.g., via AptaMCTSSequenceEncoder)
+    before passing them in here.
```
[the remark this answers is not in the document]
Comment (aditi-dsi) [reply pyaptamer/aptamcts/_feature_classifier.py]: Thanks for flagging. It got left behind while removing the encoder to use utils instead.

File: pyaptamer/aptamcts/tests/test_aptamcts.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+__author__ = ["aditi-dsi"]
+
+import numpy as np
+import pytest
+from sklearn.utils.estimator_checks import parametrize_with_checks
+
+from pyaptamer.aptamcts import AptaMCTSClassifier, AptaMCTSPipeline
+
+params = [
```
Comment (siddharth7113) [inline pyaptamer/aptamcts/tests/test_aptamcts.py:10]: Hi @aditi-dsi, thanks for this,

One bigger-picture note: we're moving the experiment wrappers onto a shared input
contract via `MoleculeLoader`
(https://github.com/gc-os-ai/pyaptamer/blob/main/pyaptamer/data/loader.py).
`AptamerEvalAptaNet` already uses it and is a good template
(https://github.com/gc-os-ai/pyaptamer/blob/main/pyaptamer/experiments/_aptamer_aptanet.py);
AptaMCTS still takes the older tuple/DataFrame input.

Comment (siddharth7113) [reply pyaptamer/aptamcts/tests/test_aptamcts.py:10]: Could you please update the PR to match it too, otherwise it would need to be re-written soon.

**Review by siddharth7113 — CHANGES_REQUESTED** [review]

File: pyaptamer/aptamcts/tests/test_aptamcts.py (hunk 2 above)
Comment (aditi-dsi) [reply pyaptamer/aptamcts/tests/test_aptamcts.py:10]: Sure, I'll update the PR.

File: pyaptamer/aptamcts/_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    ...     data={"aptamer": [aptamer_seq] * 4, "protein": [protein_seq] * 4}
+    ... )
+    >>> y_train = np.array([0, 1, 0, 1], dtype=np.float32)
+    >>> pipe.fit(X_train, y_train)  # doctest: +ELLIPSIS
+    AptaMCTSPipeline(...)
+    >>> preds = pipe.predict(X_train)
+    >>> proba = pipe.predict_proba(X_train)
+    """
+
+    def __init__(self, rna_k=4, prot_k=3, estimator=None):
```
Comment (siddharth7113) [inline pyaptamer/aptamcts/_pipeline.py]: Can you take here all the parameters of `PairsToFeatures` so that it is exposed also in the pipeline?

File: pyaptamer/aptamcts/_transforms.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
+"""Feature-extraction transforms for the AptaMCTS algorithm."""
```
Comment (siddharth7113) [inline pyaptamer/aptamcts/_transforms.py:1]: Can you add test for this? It would be a good idea to have test coverage and know if any edge case exsist?

File: pyaptamer/utils/_aptamcts_utils.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
+__author__ = "aditi-dsi"
```
Comment (siddharth7113) [inline pyaptamer/utils/_aptamcts_utils.py]: nit: this should be a list.

**Review by siddharth7113 — CHANGES_REQUESTED** [review]
Overall this looks good,

Can you add tests for `iCTF` encoders along with an end to end test for `AptamerEvalAptaMCTS` and update documentation about this algorithm , I think that it would be good to go.
